docs/CreatingClientInstance.md

```diff
@@ -0,0 +1,68 @@
+# Creating Client Instance
+
+Initialization of the Client can be done in one of below two ways
+
+## 1. Create With ClientOptions [Recommended]
+
+In order to instantiate a Client object, one has to pass in the `authProvider` or `middleware` chain in [ClientOptions](../src/IClientOptions.ts).
+
+### Option A. Default Middleware chain
+
+Pass an instance of a class which implementing [AuthenticationProvider](../src/IAuthenticationProvider.ts) interface as `authProvider` in [ClientOptions](../src/IClientOptions.ts), which will instantiate the Client with default set of middleware chain.
+
+Library is shipped with one such authentication provider named [MSALAuthenticationProvider](../src/MSALAuthenticationProvider.ts). This MSALAuthenticationProvider depends on an authentication library [msal.js](https://github.com/AzureAD/microsoft-authentication-library-for-js) which is not shipped along with the library, one has to externally include msal.js to use MSALAuthenticationProvider.
+
+```typescript
+// Instantiating Client with MSALAuthenticationProvider
+let clientOptions: ClientOptions = {
+    authProvider: new MSALAuthenticationProvider(<CLIENT_ID>, <SCOPES>, <OPTIONS>)
+};
+const client = Client.initWithMiddleware(clientOptions);
+```
+
+Want to use own preferred authentication library, for which one has to implement [AuthenticationProvider](../src/IAuthenticationProvider.ts) interface and pass in the instance of it as `authProvider` in [ClientOptions](../src/IClientOptions.ts).
+
+```typescript
+let clientOptions: ClientOptions = {
+    // MyCustomAuthenticationProvider is the user's own authentication provider implementing AuthenticationProvider interface
+    authProvider: new MyCustomAuthenticationProvider()
+};
+const client = Client.initWithMiddleware(clientOptions);
+```
+
+Refer, [custom authentication provider](./CustomAuthenticationProvider.md) for more detailed information.
+
+### Option B. Custom Middleware chain
+
+Want to have complete control over the request and the response objects, one can provide his own chain of middleware.
+Have to pass first middleware in the chain as `middleware` in [ClientOptions](../src/IClientOptions.ts).
+
+```typescript
+let clientOptions: ClientOptions = {
+    // MyFirstMiddleware is the first middleware in my custom middleware chain
+    middleware: new MyFirstMiddleware()
```

Comment:
Can i provide both middleware and authProvider together ?

Reply:
No you can't... in such case authProvider will be taken into account.

Reply:
I guess its better to throw an error in that case. will do that.

Reply:
also, add this to the documentation..